homeassistant/components/nws/__init__.py

```diff
@@ -52,38 +52,54 @@ def signal_unique_id(latitude, longitude):
 
 
 async def async_setup(hass: HomeAssistant, config: dict):
-    """Set up the National Weather Service integration."""
-    if DOMAIN not in config:
-        return True
-
-    hass.data[DOMAIN] = hass.data.get(DOMAIN, {})
-    for entry in config[DOMAIN]:
-        latitude = entry.get(CONF_LATITUDE, hass.config.latitude)
-        longitude = entry.get(CONF_LONGITUDE, hass.config.longitude)
-        api_key = entry[CONF_API_KEY]
-
-        client_session = async_get_clientsession(hass)
-
-        if base_unique_id(latitude, longitude) in hass.data[DOMAIN]:
-            _LOGGER.error(
-                "Duplicate entry in config: latitude %s  latitude: %s",
-                latitude,
-                longitude,
-            )
-            continue
-
-        nws_data = NwsData(hass, latitude, longitude, api_key, client_session)
-        hass.data[DOMAIN][base_unique_id(latitude, longitude)] = nws_data
-        async_track_time_interval(hass, nws_data.async_update, DEFAULT_SCAN_INTERVAL)
-
-        for component in PLATFORMS:
-            hass.async_create_task(
-                discovery.async_load_platform(hass, component, DOMAIN, entry, config)
-            )
+    """Set up the National Weather Service (NWS) component."""
+    hass.data.setdefault(DOMAIN, {})
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up a National Weather Service entry."""
+    latitude = entry.data[CONF_LATITUDE]
+    longitude = entry.data[CONF_LONGITUDE]
+    api_key = entry.data[CONF_API_KEY]
+    station = entry.data[CONF_STATION]
+
+    client_session = async_get_clientsession(hass)
+
+    nws_data = NwsData(hass, latitude, longitude, api_key, client_session)
+    hass.data[DOMAIN][entry.entry_id] = nws_data
+
+    # async_set_station only does IO when station is None
+    await nws_data.async_set_station(station)
+    await nws_data.async_update()
 
+    scan_time = DEFAULT_SCAN_INTERVAL
+    async_track_time_interval(hass, nws_data.async_update, scan_time)
```

Comment:
```suggestion
    async_track_time_interval(hass, nws_data.async_update, DEFAULT_SCAN_INTERVAL)
```

Not configurable so no need to copy.